### ISSUE TITLE
Error calling API handler: TypeError: rsp.getStatus is not a function

### ISSUE THREAD

**Issue by freaktechnik** [WebThingsIO/gateway#2988]
It appears this is not quite working as it should (on branch v1.0.X):

```
2022-10-16 13:18:17.914 ERROR  : Error calling API handler: TypeError: rsp.getStatus is not a function
    at /home/martin/dev/gateway/build/controllers/extensions_controller.js:65:33
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
```

https://github.com/WebThingsIO/gateway/blob/15a1b23657a832c00a02b7856cc4e0e4338ded4d/src/controllers/extensions_controller.ts#L42

**Comment by freaktechnik** [comment]
I think this is because the published/used version of `gateway-addon` doesn't have the typescript conversion yet? Though in my adapter code `getStatus` works just fine.

**Comment by freaktechnik** [comment]
Or maybe it's due to some serialization when communicationg with the plugin in https://github.com/WebThingsIO/gateway/blob/15a1b23657a832c00a02b7856cc4e0e4338ded4d/src/plugin/api-handler-proxy.ts#L49

**Comment by benfrancis** [comment]
@freaktechnik Thanks for reporting.

> I think this is because the published/used version of gateway-addon doesn't have the typescript conversion yet?

I think it does https://github.com/WebThingsIO/gateway-addon-node/tree/1.0.1-alpha.1/src

> Or maybe it's due to some serialization when communicationg with the plugin in

I'm not sure, I'm not familiar with this area of the code.

Are you able to provide steps to reproduce and/or a minimal test case?

Does this break any end user functionality that you've noticed, or just generate an error on the command line?

**Comment by bewee** [comment]
Duplicate of #2878
A simple-but-rather-dirty-fix has been suggeseted here: https://github.com/WebThingsIO/gateway/pull/2886

**Comment by benfrancis** [comment]
Thanks @bewee, closing.

**Comment by benfrancis** [comment]
Actually, let's keep this issue open to track the bug on the 1.0.X branch, until the fix from master is backported and landed there.

**Comment by benfrancis** [comment]
@bewee Since this was reported on 1.0.X too, would you mind creating a PR which applies this same patch to that branch?

**Comment by benfrancis** [comment]
Fixed on 1.0.X in https://github.com/WebThingsIO/gateway/pull/2996
